### ISSUE TITLE
dictionary KeyError using ASE OTF training

### ISSUE THREAD

**Issue by aaronchen0316** [mir-group/flare#242]
Hi I encountered an error while using ASE_OTF() to train my system. 
```
flare_calculator = FLARE_Calculator(gp_model,
                                    par = True,
                                    mgp_model = None,
                                    use_mapping = False)
cube.set_calculator(flare_calculator)
otf = ASE_OTF(cube,
                   timestep = 1 * units.fs,
                   number_of_steps = 5000,
                   dft_calc = dft_calc,
                   md_engine = md_engine,
                   md_kwargs = md_kwargs,
                   **otf_params)
```
I copied the error messages below
```
Process Process-415:
Traceback (most recent call last):
  File "/path_to_python/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/path_to_python/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "/path_to_code/code/my_flare/flare/gp_algebra.py", line 21, in queue_wrapper
    result_queue.put((wid, func(*args)))
  File "/path_to_code/code/my_flare/flare/gp_algebra.py", line 1179, in get_ky_and_hyp_pack
    training_data = _global_training_data[name]
KeyError: 'default_gp' (or whatever name='' I specified)
```
I don't know what causes this error because some training can run without any problems but some will encounter this KeyError. Also, this error didn't occur until a few steps in the training.
My best guess is that I am running my script in parallel on a shared computation node (with 16-24 cpus), so each subprocess has problem communicating?

Thank you in advance!

**Comment by YuuuuXie** [comment]
Could you please tell which version of flare are you using? I think it should have been addressed in #223

**Comment by aaronchen0316** [comment]
> Could you please tell which version of flare are you using? I think it should have been addressed in #223

My bad. My version is #218. Will check with the later version. Thank you
